sdk/src/Core/Amazon.Runtime/_pcl/ClientContext.pcl.cs

```diff
@@ -38,9 +38,10 @@ public ClientContext(string appID)
             {
                 _clientID = _appSetting.GetValue(APP_ID_KEY, ApplicationSettingsMode.Local);
                 if (string.IsNullOrEmpty(_clientID))
-                {
-                    _clientID = Guid.NewGuid().ToString();
-                    _appSetting.SetValue(APP_ID_KEY, _clientID, ApplicationSettingsMode.Local);
+                { 
+                    string newClientID = Guid.NewGuid().ToString();
+                    _appSetting.SetValue(APP_ID_KEY, newClientID, ApplicationSettingsMode.Local);
```

Comment:
Should we check if `SetValue` failed or succeeded and then log an error message when SetValue fails and not store _clientID?